tools/importer-rest-api-specs/components/dataapigeneratorjson/models/terraform_resource_tests.go

```diff
@@ -0,0 +1,16 @@
+package models
+
+// TerraformResourceTestConfig describes the various test configurations that a Resource will be tested against
+type TerraformResourceTestConfig struct {
+	// Basic is the most basic hcl that only contains the minimum attributes needed to create this specific resource
+	Basic string `json:"Basic"`
+
+	// RequiresImport is the hcl that will be used to confirm a Resource is imported correctly
+	RequiresImport string `json:"RequiresImport"`
+
+	// CompleteConfig is the most complete hcl that contains all the attributes a Resource manages
+	CompleteConfig *string `json:"CompleteConfig,omitempty"`
+
+	// TemplateConfig is the hcl that contains the various resources needed to provision this specific resource
+	TemplateConfig *string `json:"TemplateConfig,omitempty"`
+}
```

Comment:
whilst we're not using it right now, the Terraform generator supports "other" tests (e.g. run these configs) - so we should also support this - as such we'll need to include a field for this one:

```suggestion
	OtherTests *map[string][]string `json:"otherTests,omitempty"`
}
```